webapp/src/js/controllers/contacts.js

```diff
@@ -218,89 +226,92 @@ var _ = require('underscore'),
                      settings.muting.unmute_forms.includes(formId));
     };
 
-    $scope.setSelected = function(selected) {
+    $scope.setSelected = function(selected, options) {
       liveList.setSelected(selected.doc._id);
-      $scope.selected = selected;
+      ctrl.setLoadingSelectedChildren(true);
+      ctrl.setLoadingSelectedReports(true);
+      ctrl.setSelected(selected);
       ctrl.clearCancelCallback();
-      var selectedDoc = selected.doc;
       var title = '';
-      if (selected.doc.type === 'person') {
+      if (ctrl.selected.doc.type === 'person') {
         title = 'contact.profile';
       } else {
-        title = ContactSchema.get(selected.doc.type).label;
+        title = ContactSchema.get(ctrl.selected.doc.type).label;
       }
       $scope.loadingSummary = true;
       return $q
         .all([
           $translate(title),
-          getActionBarDataForChild(selectedDoc.type),
-          getCanEdit(selectedDoc),
+          getActionBarDataForChild(ctrl.selected.doc.type),
+          getCanEdit(ctrl.selected.doc),
         ])
         .then(function(results) {
           $scope.setTitle(results[0]);
           if (results[1]) {
-            selectedDoc.child = results[1];
+            ctrl.updateSelected({ doc: { child: results[1] }});
```

Comment:
Again this style of updating implies knowledge of the internal structure of the store, but in this instance I think it's ok as this is updating an object that was put into the store, not actually traversing the store itself.